### PR TITLE
Feature/core v0.2

### LINKED ISSUE #7: Update core to v0.2

### PR DESCRIPTION
**Pull request by mat-hek**
closes #7

### CODE REVIEW

File: c_src/membrane_element_ffmpeg_swresample/converter.c [hunk 1]
```diff
@@ -1,138 +1,135 @@
...
-  char* init_error = init(
-    handle,
-    from_s24le,
-    src_av_format, src_rate, src_layout,
-    dst_av_format, dst_rate, dst_layout
-  );
-  if(init_error)
-    return membrane_util_make_error_internal(env, init_error);
+  if (nb_channels_to_av_layout(src_channels, &src_layout)) {
+    err_reason = "unsupported_src_no_channels";
```
Comment (bblaszkow06) [inline c_src/membrane_element_ffmpeg_swresample/converter.c]: No channels suggests that there are no channels at all :wink: I would change it to `unsupported_src_channels_no`

File: c_src/membrane_element_ffmpeg_swresample/converter.c [hunk 2]
```diff
@@ -1,138 +1,135 @@
...
+    goto error;
+  }
+  if (nb_channels_to_av_layout(dst_channels, &dst_layout)) {
+    err_reason = "unsupported_dst_no_channels";
```
Comment (bblaszkow06) [inline c_src/membrane_element_ffmpeg_swresample/converter.c]: As above

File: c_src/membrane_element_ffmpeg_swresample/converter.c [hunk 3]
```diff
@@ -1,138 +1,135 @@
...
+    goto error;
+  }
 
-  ERL_NIF_TERM converter_term = enif_make_resource(env, handle);
-  enif_release_resource(handle);
+  state = unifex_alloc_state(env);
+  err_reason = lib_init(state, from_s24le, src_av_format, src_rate, src_layout,
+                        dst_av_format, dst_rate, dst_layout);
+  if (err_reason) {
+    goto error;
```
Comment (bblaszkow06) [inline c_src/membrane_element_ffmpeg_swresample/converter.c]: This does nothing.

File: c_src/membrane_element_ffmpeg_swresample/converter.c [hunk 4]
```diff
@@ -1,138 +1,135 @@
...
+  }
 
-  return membrane_util_make_ok_tuple(env, converter_term);
+error:
```
Comment (bblaszkow06) [inline c_src/membrane_element_ffmpeg_swresample/converter.c]: This label is confusing. Either rename it to fn_name_exit or place a happy path before the label and get rid of that if in the next line

File: c_src/membrane_element_ffmpeg_swresample/converter.c [hunk 5]
```diff
@@ -1,138 +1,135 @@
...
+  if (err_reason) {
+    if (state) {
+      unifex_release_state(env, state);
+    }
+    return create_result_error(env, err_reason);
+  } else {
+    UNIFEX_TERM res = create_result_ok(env, state);
+    unifex_release_state(env, state);
```
Comment (bblaszkow06) [inline c_src/membrane_element_ffmpeg_swresample/converter.c]: The whole point of using goto is to release resources only in one place.

File: lib/membrane_element_ffmpeg_swresample/converter.ex [hunk 6]
```diff
@@ -7,31 +7,32 @@ defmodule Membrane.Element.FFmpeg.SWResample.Converter do
...
+  def_options input_caps: [
                 type: :caps,
                 spec: Caps.t() | nil,
                 default: nil,
                 description: """
-                Audio caps for sink pad (input). If set to nil (default value),
-                caps are assumed to be received from :sink. If explicitly set to some
-                caps, they cannot be changed by caps received from :sink.
+                Audio caps for input pad. If set to nil (default value),
+                caps are assumed to be received from :input. If explicitly set to some
```
Comment (bblaszkow06) [inline lib/membrane_element_ffmpeg_swresample/converter.ex]: Put atoms into backticks

File: c_src/membrane_element_ffmpeg_swresample/converter.c (hunk 3 above)
Comment (mat-hek) [reply c_src/membrane_element_ffmpeg_swresample/converter.c]: It does now, but it wouldn't if some code is added below.

**Pull request merged**